localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/analytics.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from localstack.http import Response
+from localstack.utils.analytics.usage import UsageSetCounter
+
+from ..api import RestApiGatewayHandler, RestApiGatewayHandlerChain
+from ..context import RestApiInvocationContext
+
+LOG = logging.getLogger(__name__)
+
+
+class IntegrationUsageCounter(RestApiGatewayHandler):
+    counter: UsageSetCounter
+
+    def __init__(self, counter: UsageSetCounter = None):
+        self.counter = counter or UsageSetCounter(namespace="apigateway:invokedrest")
+
+    def __call__(
+        self,
+        chain: RestApiGatewayHandlerChain,
+        context: RestApiInvocationContext,
+        response: Response,
+    ):
+        if context.integration:
+            invocation_type = context.integration["type"]
+            if invocation_type == "AWS":
+                service_name = self._get_aws_integration_service(context.integration.get("uri"))
+                invocation_type = f"{invocation_type}:{service_name}"
+        else:
+            # if the invocation does not have an integration attached, it probably failed before routing the request,
+            # hence we should count it as a NOT_FOUND invocation
+            invocation_type = "NOT_FOUND"
```

Comment:
Question: I wonder if this is a relevant data point? On it's own, what intel can we gather from it, except for maybe knowing the total amount of requests sent to apigw? But is it relevant?

Reply:
I think it would be good to know the total amount of requests sent to the endpoint. Not very relevant directly, but I think relevant if we do the sum of all types regardless of their value?

Reply:
Maybe @thrau could chime in too, I also see the point, so yeah I think it depends on expectations!

Reply:
Btw, I am OK either way! :+1: I was simply raising a question, to make sure we are deliberate with logging :wink: